### PR TITLE
Matching: UI for item level matching #131

### PR DESCRIPTION
**Pull request by ilkkalehtinen**
-Initial item level matching UI without backend integration

### CODE REVIEW

File: src/client/components/Matching/Items/Candidates.react.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    intl: intlReducer,
+    datagrid: datagridReducer,
+  }),
+  compose(applyMiddleware(thunk)),
+);
+
+
+export default class CandidateList extends React.Component {
+  componentWillReceiveProps(nextProps) {
+    store.dispatch(DatagridActions.setData('candidates-grid',
```
Comment (alevchuk-sc) [inline src/client/components/Matching/Items/Candidates.react.js]: Dispatching redux state-changing actions in componentWillReceiveProps leads to infinite rerendering cause this method is called every time the component is provided with new props. Checking for specific condition signifying that we actually should reload candidates list would prevent such behavior.

File: src/client/components/Matching/Items/Items.react.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            type="text"
+            value=""
+            placeholder="Search"
+          />
+        </FormGroup>
+        <List
+          items={items}
+          selected={[0]}
+          multiple={false}
+          onChange={this.handleChange}
```
Comment (alevchuk-sc) [inline src/client/components/Matching/Items/Items.react.js]: Method handleChange hasn't been defined in the component.

File: src/client/components/Matching/Items/Candidates.react.js (hunk 1 above)
Comment (ilkkalehtinen) [reply src/client/components/Matching/Items/Candidates.react.js]: Not in this case. The props are passed on to datagridReducer, and those are not connected to this component.

File: src/client/components/Matching/Items/Items.react.js (hunk 2 above)
Comment (ilkkalehtinen) [reply src/client/components/Matching/Items/Items.react.js]: Fixed.

File: src/client/components/Matching/layouts/MatchingItemsLayout.react.js [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
+const store = createStore(
+  combineReducers({
+    intl: intlReducer,
+    datagrid: datagridReducer,
+    matching,
+  }),
+  compose(applyMiddleware(thunk)),
+);
+
 class MatchingItemsLayout extends React.Component {
```
Comment (alevchuk-sc) [inline src/client/components/Matching/layouts/MatchingItemsLayout.react.js]: Please make this component pure functional cause with your last changes all it's state and corresponding actions have been moved to redux store.

File: src/client/components/Matching/list/MatchingList.react.js [hunk 4]
```diff
@@ -45,11 +46,15 @@ export default class MatchingList extends PureComponent {
     }
   }
 
+  handleMatchInvoice = (invoiceId) => {
+    this.props.router.push(`${InvoiceViews.MATCHING.path}/${invoiceId}`);
```
Comment (alevchuk-sc) [inline src/client/components/Matching/list/MatchingList.react.js]: Please use router from context cause now it's not passed with component props. Also referred InvoiceViews hasn't been imported so it's undefined.

File: src/client/components/Matching/layouts/MatchingItemsLayout.react.js (hunk 3 above)
Comment (ilkkalehtinen) [reply src/client/components/Matching/layouts/MatchingItemsLayout.react.js]: Fixed

File: src/client/components/Matching/list/MatchingList.react.js (hunk 4 above)
Comment (ilkkalehtinen) [reply src/client/components/Matching/list/MatchingList.react.js]: Fixed

**Pull request merged**